### PR TITLE
Remove generic key which over-constrains query

### PR DESCRIPTION
**Pull request by ghost**
Aims: Fix bug where passcode is sometimes not accepted.

Test: 
1. Go through consent process, enter a passcode. 
2. Go to Device Settings and put the device time forward by 1 hour. 
3. Reopen the app from the background. 
4. Enter the passcode. 

Expected: 
Passcode is accepted and previously viewed tab is shown. 

Actual without fix: 
Passcode is denied

### CODE REVIEW

File: APCAppCore/APCAppCore/Library/Objects/APCKeychainStore.m [hunk 1]
```diff
@@ -191,7 +189,6 @@ + (BOOL) removeItemForKey: (NSString *) key
         NSMutableDictionary *itemToDelete = [[NSMutableDictionary alloc] init];
         [itemToDelete setObject:(__bridge id)kSecClassGenericPassword forKey:(__bridge id)kSecClass];
         [itemToDelete setObject:service forKey:(__bridge id)kSecAttrService];
-        [itemToDelete setObject:key forKey:(__bridge id)kSecAttrGeneric];
         [itemToDelete setObject:key forKey:(__bridge id)kSecAttrAccount];
 #if !TARGET_IPHONE_SIMULATOR && defined(__IPHONE_OS_VERSION_MIN_REQUIRED)
```
Comment (YuanZhu-apple) [inline APCAppCore/APCAppCore/Library/Objects/APCKeychainStore.m:193]: Can this dictionary generation code be consolidated into one place?

Comment (ghost) [inline APCAppCore/APCAppCore/Library/Objects/APCKeychainStore.m:193]: The objects set on the dictionary/query are not consistent between methods, and removeAllItemsForService uses the dictionary in a loop. The only consistent line is 
[query setObject:(__bridge id)kSecClassGenericPassword forKey:(__bridge id)kSecClass];

Comment (YuanZhu-apple) [inline APCAppCore/APCAppCore/Library/Objects/APCKeychainStore.m:193]: Would this work?

```
static NSMutableDictionary* NSMutableDictionary createQuery(id class, id service, id account) {
    ...
}
```

Comment (ghost) [inline APCAppCore/APCAppCore/Library/Objects/APCKeychainStore.m:193]: We could do that, providing we cover all the parameters for all the methods, such as kSecReturnData, kSecReturnAttributes, kSecMatchLimit, kSecAttrAccessGroup...
@insha  any comment to add?

**Comment by YuanZhu-apple** [comment]
@insha @chrisinsfo Can you address this? 
Then I can get it merged.

**Comment by insha** [comment]
While the code is working, consolidating this code will affect all our apps and therefore will take a while with testing. Is it possible that we create an issue on GitHub to track this?

**Comment by YuanZhu-apple** [comment]
OK, @insha Can you create a issue from @chrisinsfo 's comment?

**Comment by ghost** [comment]
I'm concerned that setting nil as a query parameter via a class method to create the query, may actually constrain the query, which is what I'm trying to avoid. Will need testing or advice - @YuanZhu-apple  ?

**Comment by insha** [comment]
@YuanZhu-apple Will do. Thank you.

**Comment by YuanZhu-apple** [comment]
@chrisinsfo How about we discuss it when the PR for #80 is submitted?

**Pull request merged**